omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1078,28 +1098,31 @@ def _move_hand_direct_joint(self, joint_pos, stop_on_contact=False, ignore_failu
         Returns:
             th.tensor or None: Action array for one step for the robot to move arm or None if its at the joint positions
         """
-        controller_name = f"arm_{self.arm}"
-        use_delta = self.robot._controllers[controller_name].use_delta_commands
 
         # Store the previous eef pose for checking if we got stuck
         prev_eef_pos = th.zeros(3)
 
-        for _ in range(m.MAX_STEPS_FOR_HAND_MOVE_JOINT):
+        # All we need to do here is save the target joint position so that empty action takes us towards it
+        controller_name = f"arm_{self.arm}"
+        self._arm_targets[controller_name] = joint_pos
+
+        for i in range(m.MAX_STEPS_FOR_HAND_MOVE_JOINT):
             current_joint_pos = self.robot.get_joint_positions()[self._manipulation_control_idx]
-            diff_joint_pos = th.tensor(joint_pos) - th.tensor(current_joint_pos)
+            diff_joint_pos = joint_pos - current_joint_pos
             if th.max(th.abs(diff_joint_pos)).item() < m.JOINT_POS_DIFF_THRESHOLD:
                 return
             if stop_on_contact and detect_robot_collision_in_sim(self.robot, ignore_obj_in_hand=False):
                 return
-            if th.max(th.abs(self.robot.get_eef_position(self.arm) - prev_eef_pos)).item() < 0.0001:
+            # check if the eef stayed in the same pose for sufficiently long
+            if (
+                og.sim.get_sim_step_dt() * i > m.TIME_WITHOUT_CHECKING
```

Comment:
More descriptive name for TIME_WITHOUT_CHECKING - perhaps something TIME_BEFORE_JOINT_STUCK_CHECK